### DISCUSSION TITLE
Go and Cgo

### DISCUSSION THREAD

**Discussion by yann-soubeyrand** [pulumi/pulumi#11526] [General]
Hello,

I was recently hit by an issue with a dependency of the Pulumi Go SDK:
```
    # github.com/pjbgf/sha1cd/cgo
    vendor/github.com/pjbgf/sha1cd/cgo/sha1.go:3:11: fatal error: lib/sha1.h: No such file or directory
        3 | // #include <lib/sha1.h>
          |           ^~~~~~~~~~~~
    compilation terminated.

    error: error in compiling Go: unable to run `go build`: exit status 2
```
```
❯ go mod why github.com/pjbgf/sha1cd/cgo
# github.com/pjbgf/sha1cd/cgo
<my-pulumi-go-project>
github.com/pulumi/pulumi/sdk/v3/go/pulumi
github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin
github.com/pulumi/pulumi/sdk/v3/go/common/workspace
github.com/go-git/go-git/v5/plumbing
github.com/go-git/go-git/v5/plumbing/hash
github.com/pjbgf/sha1cd/cgo
```
This issue has since been fixed. However, this has enlighten that Pulumi doesn’t disable Cgo (by setting the environment variable `CGO_ENABLED=0`) when building Go programs and I was wondering if it’s a deliberate choice or not?

**Comment by AaronFriel** [comment]
That error is surprising to me, do you get this error when using `pulumi up` and associated commands?

**Reply by yann-soubeyrand** [reply]
Yes, I get it using `pulumi up` in this repository: https://github.com/yann-soubeyrand/pulumi-cgo. The underlying issue is that the `github.com/pjbgf/sha1cd/cgo` package couldn’t be vendored because it needed C sources which were in a sub-folder of the package and Go vendoring doesn’t take sub-folders into account (this issue has since be fixed in a new release). However, when disabling Cgo, this issue isn’t a problem. This is how I discovered that Cgo is enabled (in fact isn’t disabled) when Pulumi compiles Go programs.
